### DISCUSSION TITLE
NO_RESPONSE_FROM_FUNCTION on Vercel

### DISCUSSION THREAD

**Discussion by positonic** [nextauthjs/next-auth#279] [Help] [answered]
*Please refer to the [documentation](https://next-auth.js.org/getting-started/introduction), the [example project](https://github.com/iaincollins/next-auth-example) and existing issues before creating a new issue.*

**Your question**
Has anyone tested deploying next-auth to Vercel? I 'm getting the below errors.

**What are you trying to do**
I'm trying to get next-auth working on vercel.

**Documentation feedback**
The docs are good, I'm begging for assistance as I'm stuck, and deseperate enough to jump ship.

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

I suspect it's a some kind of environmental variable issue. I have added them in Vercel, and debugging shows them coming out in the logs during build time.

[This is the page in question](https://kite-nine.vercel.app/api/auth/callback/google)
I don't see logging for page refreshes, but I'm new to NextJS and to Vercel (in addition to next-auth), so I could be doing something foolish here.

![image](https://user-images.githubusercontent.com/10499048/84955904-67ea2e80-b0f8-11ea-96fe-8d7f523dec94.png)

In the logs I see:

> [GET] /api/auth/callback/google
00:06:50:64
request",\n' +
      '  "error_description": "Missing required parameter: code"\n' +
      '}'
  },
  undefined,
  'google',
  undefined
] 
https://next-auth.js.org/errors#oauth_get_access_token_error
2020-06-17T22:06:51.787Z	331103bd-f6d2-4529-b98e-de5ba14bb74e	ERROR	[next-auth][error][OAUTH_GET_PROFILE_ERROR] [
  {
    statusCode: 401,
    data: '{\n' +
      '  "error": {\n' +
      '    "code": 401,\n' +
      '    "message": "Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",\n' +
      '    "status": "UNAUTHENTICATED"\n' +
      '  }\n' +
      '}\n'
  }
]

**Comment by iaincollins** [comment]
For reference, the example project is deployed in Vercel.

If I click the Google Sign In button at https://kite-nine.vercel.app/api/auth/signin, it redirects to Google but Google returns an error error:

![Google Error](https://user-images.githubusercontent.com/595695/84960745-5bff6c00-b0fa-11ea-9539-1bdbfd7ef11e.png)

That's an error from Google, I would check the App configuration in Google Console as it seems like the callback URL configuration in Google for the app is is wrong.

You don't want to go to the callback page directly, that's a page Google invokes as a callback as part of the sign in journey.

**Comment by positonic** [comment, marked as answer]
Oh wow it finally works after hours of pulling my hair out.

The only reason I was visiting the url directly was because I couldn't add the Vercel URL as a redirect URL in Google console. It kept saying saving... with a spinner.

I could add any random page, so I visited the url and got the undefined errors.

Thanks again for your wonderful support Iain!

**Comment by iaincollins** [comment]
Great, glad it's working! :-)

That page probably shouldn't just error like that in Google if access directly.

Happy to take any suggestions as to what it _should_ do!

(Redirect to sign for that provider?)

**Comment by abcd-ca** [comment]
@jamespfarrell I wonder if your 502 error is related to the fix I found here, https://github.com/nextauthjs/next-auth/issues/825#issuecomment-740080483
